samples/sample_decode/src/pipeline_decode.cpp

```diff
@@ -1884,6 +1897,23 @@ mfxStatus CDecodingPipeline::RunDecoding()
                 bErrIncompatibleVideoParams = true;
                 // need to go to the buffering loop prior to reset procedure
                 pBitstream = NULL;
+                sts = MFX_ERR_NONE;
+                continue;
+            } else if (MFX_ERR_REALLOC_SURFACE == sts) {
+                mfxVideoParam param;
+                m_pmfxDEC->GetVideoParam(&param);
+
+                m_pCurrentFreeSurface->frame.Info.CropW = param.mfx.FrameInfo.CropW;
+                m_pCurrentFreeSurface->frame.Info.CropH = param.mfx.FrameInfo.CropH;
+                m_YUV_Width = MSDK_ALIGN16(std::max(param.mfx.FrameInfo.Width, (mfxU16)m_YUV_Width));
+                m_YUV_Height = MSDK_ALIGN16(std::max(param.mfx.FrameInfo.Height, (mfxU16)m_YUV_Height));
+                m_pCurrentFreeSurface->frame.Info.Width = m_YUV_Width;
+                m_pCurrentFreeSurface->frame.Info.Height = m_YUV_Height;
+
+                inMid = m_pCurrentFreeSurface->frame.Data.MemId;
+                ReallocSurface(inMid, &m_pCurrentFreeSurface->frame.Info, m_pCurrentFreeSurface->frame.Data.MemType, &outMid);
```

Comment:
1. error check is missing
2. Rename the function and call as: `ReallocCurrentSurface(param.mfx.FrameInfo)`

```
mfxStatus ReallocCurrentSurface(mfxFrameInfo& info)
{
...
}
```

Reply:
Sure. I will change it. Thanks.